atlascope/core/rest/endpoints/dataset_endpoints.py

```diff
@@ -39,19 +39,13 @@ def create(self, request):
         return Response(DatasetSerializer(new_dataset_obj).data, status=status.HTTP_201_CREATED)
 
     @swagger_auto_schema(request_body=DatasetSubImageSerializer())
-    @action(detail=False, methods=['POST'])
-    def subimage(self, request):
+    @action(detail=True, methods=['POST'])
+    def subimage(self, request, pk):
         serializer = DatasetSubImageSerializer(data=self.request.data)
         serializer.is_valid(raise_exception=True)
-        original_dataset = Dataset.objects.get(id=serializer.validated_data["original_dataset_id"])
-        meta = serializer.validated_data.copy()
-        meta.pop("original_dataset_id")
-        new_dataset_obj = Dataset(
-            name=f'{original_dataset.name} Subimage',
-            metadata=meta,
-            source_dataset=original_dataset,
-            dataset_type="subimage",
-        )
-        new_dataset_obj.save()
 
-        return Response(DatasetSerializer(new_dataset_obj).data, status=status.HTTP_201_CREATED)
+        original = Dataset.objects.get(id=pk)
```

Comment:
You can just call `self.get_object()`

Reply:
Oh, neat! See b6fa31ab15767e3e801c8eea27d4e9a1e6cca257.